### DISCUSSION TITLE
单元测试所需数据的存放问题

### DISCUSSION THREAD

**Discussion by greyishsong** [XJTU-Graphics/dandelion#9] [Ideas]
目前渲染和几何部分的实验都在补充单元测试，@Eipi15926 和 @blackbird2003 都在新增测试代码。

考虑到渲染测试的参考图像、细分测试用的参考 mesh 数据量都比较大（可能达到 MiB 级别）且不属于代码或必要资源，我们是否应该将这些数据独立出去，不要将其纳入 git 的版本控制范围中？

如果是，我们把数据放在哪个路径下比较方便？如果不是，测试数据会明显影响 repo 体积导致下载不便吗？

**Comment by greyishsong** [comment]
补充：在目前尚未合并的 Loop 细分单元测试分支上，用于测试的输入模型文件和细分后的参考 mesh 信息都放在了 *test* 目录下，其中输入在 *test/test_models*，参考结果在 *test/ans/geometry*

上学期还没有这些单元测试，测试模型在单独放出的 dandelion-testset.zip 中。我更倾向这种做法的原因一是减小 git repo 体积，二是方便以后按需增加测试数据。

**Comment by blackbird2003** [comment]
确实，感觉把测试模型放在单独的dandelion-testset.zip中，在实验手册中要求在执行单元测试前把模型存放到指定目录下(如test/test_models），会更好一些。

**Comment by north-will** [comment]
赞同。感觉能通过相对路径语法访问到的路径应该都比较方便作为模型存放的路径，渲染测试里写的路径暂时是`test/ans/render`。

**Comment by greyishsong** [comment]
好，那我的规划是这样的：
- 在开发环境下，约定 *test/input* 目录存放输入数据、*test/ans* 目录存放参考结果
- 上述两个目录加入 *.gitignore* 文件，不用 git 管理
- 开发时可以随意增减输入和答案目录中的测试数据，分发时将上述两个目录统一压缩成 dandelion-testset.zip
- 渲染、几何处理、物理动画的测试数据分别放在名为 rendering/geometry/simulation 的子目录下，例如几何处理的输入在 *test/input/geometry*，参考结果在 *test/ans/geometry*
